src/content/get-started/async.md

```diff
@@ -0,0 +1,374 @@
+    ensure the request was successful (a status code of `200` means OK).
+    If successful, return the `response.body`, which contains the
+    fetched data (in this case, raw JSON). If the request failed, return
+    an informative error message.
+
+    Add these lines after the `Uri` construction within `getWikipediaArticle`:
+
+    ```dart
+    Future<String> getWikipediaArticle(String articleTitle) async {
+      final http.Client client = http.Client();
+      final Uri url = Uri.https(
+        'en.wikipedia.org',
+        '/api/rest_v1/page/summary/$articleTitle',
+      );
+      final http.Response response = await client.get(url); // Make the HTTP request
+
+      if (response.statusCode == 200) {
+        return response.body; // Return the response body if successful
+      }
+
+      // Return an error message if the request failed
+      return 'Error: Failed to fetch article "$articleTitle". Status code: ${response.statusCode}';
+    }
+    ```
+
+### Task 4: Refactor `searchWikipedia` into `runApp` and integrate the API call
+
+In Chapter 2, you added a `searchWikipedia` function. To make the code more organized and align with common Dart application patterns, you'll now rename that function to `runApp` and integrate the API call into it. This `runApp` function will house the core logic for handling the `wikipedia` command.
+
+1.  **Rename `searchWikipedia` to `runApp` and update its signature:**
+    Locate your `searchWikipedia` function and rename it to `runApp`. Also, update its signature to be `async` as it will now perform asynchronous operations.
+
+    Your `runApp` function should now look like this (initial part):
+
+    ```dart
+    // ... (your existing main function)
+
+    void runApp(List<String>? arguments) async { // Renamed from searchWikipedia and added 'async'
+      late String? articleTitle;
+
+      // If the user didn't pass in arguments, request an article title.
+      if (arguments == null || arguments.isEmpty) {
+        print('Please provide an article title.');
+        articleTitle = stdin.readLineSync(); // Await input from the user
+        // We'll add error handling for null input here in a moment
+      } else {
+        // Otherwise, join the arguments into the CLI into a single string
+        articleTitle = arguments.join(' ');
+      }
+
+      // ... rest of the function
+    }
+
+    // ... (your existing printUsage() function)
+    ```
+    * `void runApp(List<String>? arguments) async`: Notice that the function is now `async`. This is essential because it will call `getWikipediaArticle`, which is an `async` function itself and we'll need to `await` its result.
+
+2.  **Add `null` and empty string checks for user input:**
+    Inside `runApp`, refine the `if` block that handles the case where no arguments are provided. If `stdin.readLineSync()` returns `null` (e.g., if the user presses Ctrl+D/Ctrl+Z) or an empty string, print a message and exit the function.
```

Comment:
```suggestion
    Inside `runApp`, refine the `if` block that handles the case where no arguments are provided. If `stdin.readLineSync()` returns `null` (for example, if the user presses Ctrl+D/Ctrl+Z) or an empty string, print a message and exit the function.
```

This looks like it might all be on one line. If it is, can you break it up? I see a few of these in this review, so I would go through the whole thing and update. You'll want no more than 80 characters per line.

Reply:
I think it's just this view. This looks correct in my editor.